BlueBrain/Search
# In-detail parsing of the PMC XML paragraphs

Fixes #427.

## Description

The original motivation was to discard any LaTeX code that otherwise ended up in the paragraph text.

However, it also became clear that there is a number of other sub-elements in the paragraphs that need to be carefully processed. See #427 and the comments therein for more information.

## How to test?
Run
```bash
pytest tests/database/test_article.py::TestPubmedXMLArticleParser
```
to make sure the unit tests still work.

Additionally, you can used the code snippet in https://github.com/BlueBrain/Search/pull/437#issuecomment-915388992 and visually check the output parse.

## THE BIG QUESTION
is if we should handle `<inline-formula>` and `<disp-formula>` better than just replacing them by a placeholder. We might need the potential data inside, but parsing could be a nightmare. Potential for a new issue.

## Checklist

  (if it is not the case, please create an issue first).
  (if needed)
  (if needed)
  (if needed)
  (if a function is added or modified)
